ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/helpers/RequestHelper.java

```diff
@@ -41,22 +36,29 @@ public static String getNameRootUnderscores(final String name) {
         return Helper.camelToUnderscore(getNameRoot(name));
     }
 
-    public static boolean hasResponsePayload(final ImmutableList<Ds3ResponseCode> responseCodes) {
-        for (final Ds3ResponseCode responseCode : responseCodes) {
-            final int rc = responseCode.getCode();
-            if (rc >= 200 && rc < 300)
-            for (final Ds3ResponseType responseType : responseCode.getDs3ResponseTypes()) {
-                if (ConverterUtil.hasContent(responseType.getType())
-                    && !Objects.equals(responseType.getType(), "null")) {
-                    LOG.debug("response type " + responseType.getType());
-                    return true;
-                } else if (ConverterUtil.hasContent(responseType.getComponentType())
-                           && !Objects.equals(responseType.getComponentType(), "null")) {
-                    LOG.debug("response type " + responseType.getComponentType());
-                    return true;
-                }
-            }
+    public static String getAmazonS3InitParams(final ImmutableMap<String, String> requiredArgs) {
+        if (!requiredArgs.containsKey("bucketName")) {
```

Comment:
Instead of losing the abstraction I would have made a Map<String, Arguments> where the name points to the argument.  If anything it makes it easier to understand what that data is instead of Map<String, String> which could be anything.  Functionally they probably are the same.